Sources/SwiftParser/Statements.swift

```diff
@@ -187,13 +187,14 @@ extension Parser {
     var loopProgress = LoopProgressCondition()
     repeat {
       let condition = self.parseConditionElement()
-      let unexpectedBeforeKeepGoing: RawUnexpectedNodesSyntax?
+      var unexpectedBeforeKeepGoing: RawUnexpectedNodesSyntax? = nil
+      if let equalOperator = self.consumeIfContextualPunctuator("=="), let falseKeyword = self.consume(if: .keyword(.false)) {
+        unexpectedBeforeKeepGoing = RawUnexpectedNodesSyntax([equalOperator, falseKeyword], arena: self.arena)
+      }
       keepGoing = self.consume(if: .comma)
       if keepGoing == nil, let andOperator = self.consumeIfContextualPunctuator("&&") {
         unexpectedBeforeKeepGoing = RawUnexpectedNodesSyntax([andOperator], arena: self.arena)
```

Comment:
Wouldn't this break the assumption that `UnexpectedNodesSyntax` contains only two tokens, which is necessary to diagnose `== false`?

Reply:
Yes, it would, but round-trippability of the source file is a hard requirement we cannot break. And to be frank, I think it’s fairly unlikely that somebody writes the example above, so I don’t care too much. Having said that, if you do find a good way to produce good diagnostics for the case, I wouldn’t object.

Reply:
Fixing this particular case seems not trivial to me, so since `UnexpectedNodesSyntax` can contain more than two tokens, should the function `twoTokens` do not check this requirement?

Reply:
Or should I just let the parser produce `unexpected code '== false &&' in 'if' statement` & `expected ',' in 'if' statement`

Reply:
I think producing `unexpected code '== false &&' in 'if' statement` & `expected ',' in 'if' statement` would be good enough for now.